src/utils.py

```diff
@@ -0,0 +1,54 @@
+import os
+import re
+from semver import Version
+
+
+def get_required_packages_from_env_in(file_path) -> set[str]:
+    required_packages = set()
+    if not os.path.exists(file_path):
+        return required_packages
+    with open(file_path, 'r') as f:
+        lines = f.readlines()
+        required_packages = {line.strip().split("::")[1].split("[")[0] for line in lines if line.startswith('conda-forge::')}
```

Comment:
Please use one of the Conda APIs to read this file. See how we use `RequirementsSpec` as an example.

Conda spec files support various formats, so my worry with custom parsing is that we may have to revisit this code in future unless we rely on *official* APIs.